pandas/core/indexing.py

```diff
@@ -1818,6 +1816,9 @@ def error():
             except:
                 error()
 
+            if not ax.contains(key):
```

Comment:
is it possible to add a test for this (maybe compare the exception message)?

Reply:
Unfortunately comparing the message is not enough (it didn't change - and is already tested). We would have to check the traceback, and I have no idea of how to do it - and _if_ we want to do it.